src/SFA.DAS.EmployerFinance/Application/Commands/ProcessLevyDeclarations/ProcessLevyDeclarationsCommandHandler.cs

```diff
@@ -0,0 +1,27 @@
+using System.Threading;
+using System.Threading.Tasks;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+using SFA.DAS.EmployerFinance.Data;
+using SFA.DAS.EmployerFinance.Models;
+
+namespace SFA.DAS.EmployerFinance.Application.Commands.ProcessLevyDeclarations
+{
+    public class ProcessLevyDeclarationsCommandHandler : AsyncRequestHandler<ProcessLevyDeclarationsCommand>
+    {
+        private readonly EmployerFinanceDbContext _db;
+
+        public ProcessLevyDeclarationsCommandHandler(EmployerFinanceDbContext db)
+        {
+            _db = db;
+        }
+        
+        protected override async Task Handle(ProcessLevyDeclarationsCommand request, CancellationToken cancellationToken)
+        {
+            var accountPayeSchemes = await _db.AccountPayeSchemes.AsNoTracking().ToListAsync(cancellationToken);
+            var saga = new LevyDeclarationSaga(request.PayrollPeriod, accountPayeSchemes);
+
+            _db.LevyDeclarationSagas.Add(saga);
```

Comment:
As this command handler is invoked by the `ProcessLevyDeclarationsJob` in the Jobs project, don't we need to include the `EntityFrameworkCoreUnitOfWorkRegistry` in the Jobs project's `IoC.cs`, otherwise the changes won't get persisted?

Also, don't we need to add one of the NServiceBus UnitOfWork registry's for the `Publish()` call in `LevyDeclarationSaga`'s constructor to actually publish?

Reply:
The command handler should be invoked within the `SFA.DAS.EmployerFinance.MessageHandlers` process, the unit of work should also be in place there.